src/overlays/actors/ovl_En_Wiz/z_en_wiz.c

```diff
@@ -30,199 +87,1467 @@ const ActorInit En_Wiz_InitVars = {
+            Matrix_Translate(0.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+            Matrix_MultVec3f(&staffFlamePos, &flamePos);
+            flamePos.x += randPlusMinusPoint5Scaled(4.0f);
+            flamePos.y += randPlusMinusPoint5Scaled(7.0f);
+            flamePos.z += randPlusMinusPoint5Scaled(5.0f);
+            func_800B3030(play, &flamePos, &gZeroVec3f, &gZeroVec3f, ((Rand_ZeroFloat(1.0f) * 50.0f) + 70.0f), 10, 1);
+            SoundSource_PlaySfxAtFixedWorldPos(play, &flamePos, 10, NA_SE_EN_EXTINCT);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A455C4.s")
+        if ((limbIndex >= WIZROBE_LIMB_RIGHT_FOOT) && (this->timer == 0)) {
+            this->timer = 4;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A456A0.s")
+    if ((limbIndex == WIZROBE_LIMB_PELVIS) || (limbIndex == WIZROBE_LIMB_TORSO) ||
+        (limbIndex == WIZROBE_LIMB_LEFT_UPPER_ARM) || (limbIndex == WIZROBE_LIMB_LEFT_FOREARM) ||
+        (limbIndex == WIZROBE_LIMB_RIGHT_UPPER_ARM) || (limbIndex == WIZROBE_LIMB_RIGHT_FOREARM) ||
+        (limbIndex == WIZROBE_LIMB_NECK) || (limbIndex == WIZROBE_LIMB_HEAD) || (limbIndex == WIZROBE_LIMB_JAW) ||
+        (limbIndex == WIZROBE_LIMB_LEFT_SHIN) || (limbIndex == WIZROBE_LIMB_RIGHT_SHIN) ||
+        (limbIndex == WIZROBE_LIMB_LOINCLOTH)) {
+        Matrix_MultZero(&this->bodyPartsPos[this->bodyPartsPosIndex]);
+        this->bodyPartsPosIndex++;
+        if (this->bodyPartsPosIndex >= ARRAY_COUNT(this->bodyPartsPos)) {
+            this->bodyPartsPosIndex = 0;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A45CD8.s")
+void EnWiz_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnWiz* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A460A4.s")
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A46280.s")
+    func_8012C28C(play->state.gfxCtx);
+    func_8012C2DC(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wiz/func_80A462F8.s")
+    if ((this->action == EN_WIZ_ACTION_BURST_INTO_FLAMES) || (this->alpha != 255)) {
+        Scene_SetRenderModeXlu(play, 1, 2);
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetEnvColor(POLY_XLU_DISP++, 0, 0, 0, this->alpha);
+        POLY_XLU_DISP =
+            SkelAnime_DrawFlex(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                               NULL, EnWiz_XluPostLimbDraw, &this->actor, POLY_XLU_DISP);
+    } else {
+        Scene_SetRenderModeXlu(play, 0, 1);
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, 255, 255, 255, this->alpha);
+        SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                              NULL, EnWiz_OpaPostLimbDraw, &this->actor);
```

Comment:
Hmm, I'd slightly lead towards having the Opa/Xlu at the end to match this function, although this is probably the only case where I'd consider that so maybe not?

Reply:
Giant and Yb do what you're asking here (put Opa/Xlu at the end), and no code does it like I'm doing here. I'll swap over to maintain consistency.